### PR TITLE
[17.0][OU-ADD] new upgrade analysis

### PR DESCRIPTION
**Pull request by hbrunn**
I'm a little bit confused by the general log, and that we didn't have analysis files for the localizations before, but the rest looks good and especially the updated mail analysis will be helpful.

As I've not much experience with `upgrade_analysis`, here what I'm doing in case it's something stupid:

```bash
#!/bin/sh
set -e
echo running buildouts
env/bin/buildout -qqq -c buildout-analysis16.cfg
env3.11/bin/buildout -qqq -c buildout-analysis17.cfg
DB=openupgrade_analysis
PORT=42422
dropdb --if-exists ${DB}16
dropdb --if-exists ${DB}17
git -C parts/analysis17.0_scripts rm --quiet $(cd parts/analysis17.0_scripts && ls openupgrade_scripts/scripts/*/*/noupdate_changes.xml openupgrade_scripts/scripts/*/*/upgrade_analysis.txt)
ODOO_PID=$!
echo creating ${DB}17
bin/start_analysis17 -d ${DB}17 --without-demo=all -i upgrade_analysis --stop-after-init
echo generating upgrade records in ${DB}17
bin/python_analysis17 -c "
session.open(db='${DB}17')
install_wizard = session.env['upgrade.install.wizard'].create({})
install_wizard.select_odoo_modules()
install_wizard.install_modules()
session.env.cr.commit()
session.close()
session.open(db='${DB}17')
session.env['upgrade.generate.record.wizard'].create({}).generate()
session.env.cr.commit()
session.close()
"
echo creating ${DB}16
bin/start_analysis16 -d ${DB}16 --without-demo=all -i upgrade_analysis --stop-after-init
echo generating upgrade records in ${DB}16
bin/python_analysis16 -c "
session.open(db='${DB}16')
install_wizard = session.env['upgrade.install.wizard'].create({})
install_wizard.select_odoo_modules()
install_wizard.install_modules()
session.env.cr.commit()
session.env['upgrade.generate.record.wizard'].create({}).generate()
session.env.cr.commit()
session.close()
"
bin/start_analysis16 --http-port $PORT -d ${DB}16 &
echo running analysis ${DB}17
bin/python_analysis17 -c "
session.open(db='${DB}17')
config = session.env['upgrade.comparison.config'].create({
    'database': '${DB}16',
    'port': $PORT,
})
analysis_wizard = session.env['upgrade.analysis'].create({
    'config_id': config.id,
})
# without this, the upgrade_path == False, why?
analysis_wizard._compute_upgrade_path()
analysis_wizard.analyze()
print(analysis_wizard.log)
session.env.cr.commit()
session.close()
"
kill -s KILL $ODOO_PID
dropdb ${DB}16
dropdb ${DB}17
git -C parts/analysis17.0_scripts add .
git -C parts/analysis17.0_scripts status
git -C parts/analysis17.0_scripts commit -m '[OU-UPD] new analysis '$(date +%F)
```

(edit: updated to the final working version of the script in case anyone wants to reproduce this)

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@hbrunn the l10n_ changes are correct, because most of them have evolved since the latest upgrade, but there's a trick: Odoo has changed the module version in some of them, so that creates with our upgrade analysis a totally whole new file. It's the case for example of `l10n_at`, which was at 17.0.3.1 when it was analyzed the other time: https://github.com/StefanRijnhart/OpenUpgrade/blob/18f33f0f536f514a42767e81ff4523c88f94a8ea/openupgrade_scripts/scripts/l10n_at/17.0.3.1/upgrade_analysis.txt, and now it's at 17.0.3.2.1: https://github.com/OCA/OpenUpgrade/blob/6e9313be1cee7aed8c174132f6ec7c05d15d7c00/openupgrade_scripts/scripts/l10n_at/17.0.3.2.1/upgrade_analysis.txt

As they hasn't been put the migration scripts yet, we should delete the old migration folders IMO, so now in this PR, they will appear as renamed and only with the differences.

Are you able to do it these days, or should I do it?

**Comment by hbrunn** [comment]
thanks for the elucidation, I can delete the old folders tonight

File: openupgrade_scripts/scripts/base/17.0.1.3/noupdate_changes.xml [hunk 1]
```diff
@@ -13,17 +13,18 @@
     <field name="vat_label">GST/HST number</field>
   </record>
   <record id="ec" model="res.country">
+    <field name="address_format" eval="'%(street)s\n%(street2)s\n%(city)s\n%(country_name)s'"/>
     <field name="zip_required">0</field>
   </record>
-  <!-- <record id="es" model="res.country">
```
Comment (hbrunn) [inline openupgrade_scripts/scripts/base/17.0.1.3/noupdate_changes.xml:18]: reminder to keep this commentd

File: openupgrade_scripts/scripts/base/17.0.1.3/noupdate_changes.xml [hunk 2]
```diff
@@ -13,17 +13,18 @@
...
+  <record id="es" model="res.country">
     <field eval="'%(street)s\n%(street2)s\n%(zip)s %(city)s\n%(state_name)s\n%(country_name)s'" name="address_format"/>
-  </record> -->
+  </record>
   <record id="europe" model="res.country.group">
     <field name="name">European Union</field>
   </record>
-  <!-- <record id="main_company" model="res.company">
```
Comment (hbrunn) [inline openupgrade_scripts/scripts/base/17.0.1.3/noupdate_changes.xml]: this too

File: openupgrade_scripts/scripts/mail/17.0.1.15/upgrade_analysis.txt [hunk 3]
```diff
@@ -256,22 +256,21 @@ DEL ir.model.access: mail.access_mail_tracking_value_user
...
-NEW ir.model.constraint: mail.constraint_discuss_channel_rtc_session_channel_member_unique
-NEW ir.model.constraint: mail.constraint_discuss_channel_uuid_unique
-NEW ir.model.constraint: mail.constraint_discuss_gif_favorite_user_gif_favorite
-NEW ir.model.constraint: mail.constraint_mail_alias_domain_bounce_email_uniques
-NEW ir.model.constraint: mail.constraint_mail_alias_domain_catchall_email_uniques
-NEW ir.model.constraint: mail.constraint_mail_partner_device_endpoint_unique
+DEL ir.model.constraint: mail.constraint_bus_presence_partner_or_guest_exists
+DEL ir.model.constraint: mail.constraint_mail_activity_check_res_id_is_set
 DEL ir.model.constraint: mail.constraint_mail_alias_alias_unique
+DEL ir.model.constraint: mail.constraint_mail_blacklist_unique_email
```
Comment (hbrunn) [inline openupgrade_scripts/scripts/mail/17.0.1.15/upgrade_analysis.txt]: something goes wrong here, this one exists in both v16 and v17

**Comment by hbrunn** [comment]
hang on here, I'm regenerating this as I indeed did something stupid before

**Comment by hbrunn** [comment]
okay, this was a bumpy ride, but now I think the analysis is ~~nearly~~ correct. ~~The constraint definitions don't come
through yet.~~

Please review https://github.com/OCA/server-tools/pull/3090

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**